features/ultils/credentials.ts

```diff
@@ -0,0 +1,137 @@
+import { KintoneRestAPIClient } from "@kintone/rest-api-client";
+import type { Credential, Permission } from "./types";
+import fs from "fs";
+import path from "path";
+import Ajv from "ajv";
+import type { ErrorObject } from "ajv";
+
+type Record = {
+  key: {
+    type: "SINGLE_LINE_TEXT";
+    value: string;
+  };
+  app_id: {
+    type: "SINGLE_LINE_TEXT";
+    value: string;
+  };
+  api_tokens: {
+    type: "SUBTABLE";
+    value: Array<{
+      id: string;
+      value: {
+        token: {
+          type: "SINGLE_LINE_TEXT";
+          value: string;
+        };
+        permissions: {
+          type: "CHECK_BOX";
+          value: Permission[];
+        };
+      };
+    }>;
+  };
+};
+
+const e2eCredentialFileName = ".e2e-credentials.json";
+const e2eCredentialFilePath = path.join(
+  __dirname,
+  `../../${e2eCredentialFileName}`,
+);
+
+export const fetchCredentials: () => Promise<Credential[]> = async () => {
+  if (!process.env.CI && fs.existsSync(e2eCredentialFilePath)) {
+    return fetchFromFile();
+  }
+
+  return fetchFromKintone();
+};
+
```

Comment:
You can mask credentials on JavaScript using `core.setSecret()` of `@actions/core`.

https://github.com/actions/toolkit/tree/master/packages/core#setting-a-secret

You can also achieve this by outputting `::add-mask::` to stdout, but it can be unstable.
So I recommend to use `core.setSecret()`.

```typescript
process.stdout.write("::add-mask::" + cmd.toString() + os.EOL)
```

```suggestion
export const fetchCredentials: () => Promise<Credential[]> = async () => {
  const credentials = fs.existsSync(e2eCredentialFilePath) ? fetchFromFile() : fetchFromKintone();
  
  if (process.env.GITHUB_ACTIONS){
    maskCredentials(credentials)
  }

  return credentials;
};

const maskCredentials = (credentials: Credential[]) => {
credentials.forEach((credential) => {
  credential.api_tokens.value.forEach((apiToken) => {
    if (apiToken.value.token.value) {
      console.log(apiToken.value.token.value);
    }
  });
});
}

```

Reply:
@hung-cybo 
Sorry, this review is outdated.